adobe/aepsdk-analytics-android
# (2.0.1 release) staging -> main

## Description

In this release:
- Updated kotlin settings to min 1.4 to enable apps targeting lower versions

Internal updates:
- Analytics shared state is made available on bootup if there are any previously stored identifiers
- Shared state is only updated when an identifier is set and not on get requests 
- New tests were added for increased code coverage
- Updates on github templates and API docs

## Related Issue






## Motivation and Context



## How Has This Been Tested?





Automation: unit, functional tests pass on CI

Manual testing with test app: 
- set/clear identifiers, check bootup shared state (shared state 0)
- integration testing with Identity - set custom identifiers (syncIdentifiers), check they are included in upcoming hit
- test with Assurance, events for set/get identifiers workflow look good:
    - paired and non-paired response events are being dispatched bot both set/get workflows
    - shared state is not updated on set calls
- test with app targeting kotlin 1.4

## Screenshots (if appropriate):

## Types of changes




## Checklist:
